### ISSUE TITLE
"event malformed: event JSON missing type key" during initial sync

### ISSUE THREAD

**Issue by turt2live** [matrix-org/sliding-sync#263]
```
ss_sync  | 21:53:56 ERR V2: failed to initialise room error="events malformed: event JSON missing type key" room=!BAXLHOFjvDKUeLafmO:matrix.org state=8231
```

presumably it's checking for `""`, which is actually a valid event type (albeit silly).

**Comment by DMRobertson** [comment]
Which version of the proxy are you running?

Current logic accounts for `""`: https://github.com/matrix-org/sliding-sync/blob/066327d407faab80f9be4bd691593d0cb28402d1/state/event_table.go#L60-L62

**Comment by DMRobertson** [comment]
Suspect fixed in https://github.com/matrix-org/sliding-sync/pull/224

**Comment by turt2live** [comment]
Which version of the proxy are you running?

0.99.4

**Comment by DMRobertson** [comment]
Thanks, that's reassuring as #224 landed in v0.99.5.

Cross-referencing the event with matrix.org's DB, I can see that there's an event with type = `""`. So I'm pretty confident that this should be fixed going forward. Thanks for spotting.

**Comment by turt2live** [comment]
I don't know how I managed to deploy 0.99.4 yesterday when 0.99.5 has clearly been there for 2 weeks already
